### PR TITLE
[#34] add support for shebang

### LINKED ISSUE #34: Breaks with shebang
Prettier v2.2.1
@trivago/prettier-plugin-sort-imports v1.4.4
macOS 11.2.1

Screenshot 2021-03-07 at 4 12 34 PM

### PR DESCRIPTION
**Pull request by byara**
Closes #34

### CODE REVIEW

File: src/utils/get-code-from-ast.ts [hunk 1]
```diff
@@ -10,7 +10,11 @@ import { newLineCharacters } from '../constants';
  * @param nodes all imports
  * @param originalCode
  */
-export const getCodeFromAst = (nodes: Statement[], originalCode: string) => {
+export const getCodeFromAst = (
+    nodes: Statement[],
+    originalCode: string,
+    interpreter: InterpreterDirective | null | undefined,
```
Comment (ayusharma) [inline src/utils/get-code-from-ast.ts:16]:
```suggestion
    interpreter?: InterpreterDirective
```
May be undefined or ImperativeDirective

File: src/utils/get-code-from-ast.ts [hunk 2]
```diff
@@ -23,12 +27,16 @@ export const getCodeFromAst = (nodes: Statement[], originalCode: string) => {
...
+        ? codeWithoutImportDeclarations.replace(`#!${interpreter.value}`, '')
+        : codeWithoutImportDeclarations;
+
     const newAST = file({
         type: 'Program',
         body: nodes,
         directives: [],
         sourceType: 'module',
-        interpreter: null,
+        interpreter: interpreter ? interpreter : null,
```
Comment (ayusharma) [inline src/utils/get-code-from-ast.ts:39]: Can we make interpreter undefined?

**Review by ayusharma — CHANGES_REQUESTED** [review]
Small improvements

File: src/utils/get-code-from-ast.ts (hunk 2 above)
Comment (byara) [reply src/utils/get-code-from-ast.ts]: Yes. The type definition is:
```typescript
interface Program extends BaseNode {
    type: "Program";
    body: Array<Statement>;
    directives: Array<Directive>;
    sourceType: "script" | "module";
    interpreter?: InterpreterDirective | null;
    sourceFile: string;
}
```

**Review by ayusharma — APPROVED** [review]

**Pull request merged**